Paperwork 1.2.4

symphorien opened the pull request:
###### Motivation for this change
update (most of these updates could be made standalone. tell me if you want to split.)

###### Things done

The new home for paperwork is the gnome gitlab instance, but they have an unusual namespacing. Instead of the "traditional" owner/repo url, there are three levels: group/owner/repo: https://gitlab.gnome.org/World/OpenPaperwork/paperwork

I have adapted `fetchFromGitLab` to take an optional `group` argument.

I have checked that I can still scan/ocr succesfully with paperwork.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`) (just paperwork)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @aszlig as maintainer.

symphorien committed:
pythonPackages.pyocr: 0.4.7 -> 0.5.3

symphorien committed:
pythonPackages.pypillowfight: 0.2 -> 0.2.4

symphorien committed:
pythonPackages.{pyocr,pypillowfight}: fix homepage

aszlig commented on pkgs/top-level/all-packages.nix:
```diff
@@ -288,13 +288,13 @@ with pkgs;
-  } // removeAttrs args [ "domain" "owner" "repo" "rev" ]) // { inherit rev; };
+    url = "https://${domain}/api/v4/projects/${lib.optionalString (group != null) group+"%2F"}${owner}%2F${repo}/repository/archive.tar.gz?sha=${rev}";
+    meta.homepage = "https://${domain}/${lib.optionalString (group != null) group+"/"}${owner}/${repo}/";
+  } // removeAttrs args [ "domain" "owner" "group" "repo" "rev" ]) // { inherit rev; };
```
Can you please split this into a separate commit?

ryantm replied:
This breaks `fetchFromGitLab`.

Before:
`https://gitlab.com/api/v4/projects/psmisc%2Fpsmisc/repository/archive.tar.gz?sha=v23.2`
After:
`https://gitlab.com/api/v4/projects/%2Fpsmisc%2Fpsmisc/repository/archive.tar.gz?sha=v23.2`

ryantm replied:
@azlig I'd like to revert this PR until this is resolved, okay?

aszlig replied:
@ryantm: Ah, this is just a grouping issue, going to fix that instead.

ryantm replied:
More context https://github.com/NixOS/nixpkgs/pull/46977#issuecomment-424722471

aszlig replied:
Fixed in 593062ddc2104cf91b1cf9cc0e4f4b1437f59f6b.

aszlig reviewed (approved):
Looks good to me, thanks for the cleanups.

Given that the backend and frontend are now in one repositories it would probably make more sense to split out the source into a separate Nix expression file (eg. `source.nix`) and directly use `src = "${source}/paperwork-backend";` so that we don't need to set the source root anymore.

aszlig commented:
Tested this also on NixOS, so I ticked off the checkboxes accordingly.

@symphorien: Can you please ping me again after you've split off the `fetchFromGitLab`?

symphorien committed:
fetchFromGitLab: add a group optional argument

For example, paperwork is at https://gitlab.gnome.org/World/OpenPaperwork/paperwork
and you can fetch this repo with group = "World", owner =
"OpenPaperwork", and repo = "paperwork".

symphorien committed:
paperwork: 1.2.2 -> 1.2.4

symphorien commented:
@aszlig I just split the first commit as requested.
Regarding paperwork-src: the problem is that it does not belong either to nixpkgs top-level nor to pythonPackages. So I somewhat prefer the current situation. But if you prefer I can move the source to its own attribute.
